src/packs/checker.rs

```diff
@@ -339,3 +364,199 @@ fn get_checkers(
         }),
     ]
 }
+
+fn remove_reference_to_dependency(
+    configuration: &Configuration,
+    package_name: &str,
+    dependency_name: &str,
+) {
+    let existing_pack: &Pack =
+        configuration.pack_set.for_pack(&package_name).unwrap();
+    let file_name = format!(
+        "{}/{}/package.yml",
+        configuration.absolute_root.display(),
+        existing_pack.relative_path.display()
+    );
+    let content = fs::read_to_string(&file_name).unwrap();
+
+    let new_content_list =
+        remove_lines_with_dependency_formatted(&content, dependency_name);
+    let new_content = format!("{}\n", new_content_list.join("\n"));
+
+    fs::write(&file_name, new_content)
+        .expect("Should have been able to write the file");
+}
+
+fn remove_lines_with_dependency_formatted(
```

Comment:
This method is longer that I'd like.

Reply:
Since `write_pack_to_disk` already does this, I think we can:
- remove this function 
- change the tests since we won't need to double test serialization

I think you could just add a single integration test (that calls the CLI, I think it'd go in `tests/check_unnecessary_dependencies.rs`) that uses auto correct. One note is you'll probably need to update `set_up_fixtures` to revert what the code does so that the test does not affect the file system except temporarily.

Reply:
Thanks for looking this over Alex!